docs/src/literate/transient_heat_equation.jl

```diff
@@ -0,0 +1,205 @@
+∂Ω = union(getfaceset.((grid, ), ["top", "bottom"])...)
+dbc = Dirichlet(:u, ∂Ω, (x, t) -> t*(max_temp/T))
+add!(ch, dbc)
+close!(ch)
+update!(ch, 0.0);
+
+# ### Assembling the linear system
+# As in the heat equation example we define a `doassemble!` function that assembles the diffusion parts of the equation:
+function doassemble!(cellvalues::CellScalarValues{dim}, K::SparseMatrixCSC, dh::DofHandler, Δt::Real) where {dim}
+
+    n_basefuncs = getnbasefunctions(cellvalues)
+    Ke = zeros(n_basefuncs, n_basefuncs)
+    fe = zeros(n_basefuncs)
+
+    f = zeros(ndofs(dh))
+    assembler = start_assemble(K, f)
+
+    @inbounds for cell in CellIterator(dh)
+
+        fill!(Ke, 0)
+        fill!(fe, 0)
+
+        reinit!(cellvalues, cell)
+
+        for q_point in 1:getnquadpoints(cellvalues)
+            dΩ = getdetJdV(cellvalues, q_point)
+
+            for i in 1:n_basefuncs
+                v  = shape_value(cellvalues, q_point, i)
+                ∇v = shape_gradient(cellvalues, q_point, i)
+                fe[i] += v * dΩ
+                for j in 1:n_basefuncs
+                    ∇u = shape_gradient(cellvalues, q_point, j)
+                    Ke[i, j] += (∇v ⋅ ∇u) * dΩ
+                end
+            end
+        end
+
+        assemble!(assembler, celldofs(cell), fe, Ke)
+    end
+    return K, f
+end
+#md nothing # hide
+# In addition to the diffusive part, we also need a function that assembles the mass matrix `M`.
+function doassemble!(cellvalues::CellScalarValues{dim}, M::SparseMatrixCSC, dh::DofHandler) where {dim}
+
+    n_basefuncs = getnbasefunctions(cellvalues)
+    Me = zeros(n_basefuncs, n_basefuncs)
+
+    assembler = start_assemble(M)
+
+    @inbounds for cell in CellIterator(dh)
+
+        fill!(Me, 0)
+
+        reinit!(cellvalues, cell)
+
+        for q_point in 1:getnquadpoints(cellvalues)
+            dΩ = getdetJdV(cellvalues, q_point)
+
+            for i in 1:n_basefuncs
+                v  = shape_value(cellvalues, q_point, i)
+                for j in 1:n_basefuncs
+                    u = shape_value(cellvalues, q_point, j)
+                    Me[i, j] += (v ⋅ u) * dΩ
+                end
+            end
+        end
+
+        assemble!(assembler, celldofs(cell), Me)
+    end
+    return M
+end
+#md nothing # hide
+# ### Solution of the system
+# We first assemble all parts in the prior allocated `SparseMatrixCSC`.
+K, f = doassemble!(cellvalues, K, dh, Δt)
+M = doassemble!(cellvalues, M, dh)
+A = (Δt .* K) + M;
+# Now, we need to save all boundary condition related values of the unaltered system matrix `A`, which is done
+# by `get_rhs_data`. The function returns a `RHSData` struct, which contains all needed informations to apply
+# the boundary conditions solely on the right-hand-side vector of the problem.
+rhsdata = get_rhs_data(ch, A);
+# We initialize the previous time step, denoted by uₙ,  to $\mathbf{0}$.
```

Comment:
We just set the initial temperature to 0, right? Previous time step sounds like we care what happened before.